sdk/iot/Azure.Iot.Hub.Service/src/DevicesClient.cs

```diff
@@ -34,15 +32,15 @@ protected DevicesClient()
         /// <summary>
         /// Initializes a new instance of DevicesClient.
         /// <param name="devicesRestClient"> The REST client to perform device, device twin, and bulk operations. </param>
-        /// <param name="queryRestClient"> The REST client to perform query operations for the device. </param>
+        /// <param name="queryClient"> The convenience layer query client to perform query operations for the device. </param>
```

Comment:
Note that this is a bit different than usual. Instead of passing in the protocol layer query client, we will pass in the convenience layer query client. Since the convenience layer query client has all the pagination logic centralized, we can remove a significant chunk of code below